### ISSUE TITLE
resource config is not respected in head_start_ray_commands in cluster.yaml

### ISSUE THREAD

**Issue by xwjiang2010** [ray-project/ray#18097]
### What is the problem?

Pretty recent master. --num-cpus and --num-gpus are not respected in head_start_ray_commands in cluster.yaml.
They are overwritten by RAY_OVERRIDE_RESOURCES.

### Reproduction (REQUIRED)

`ray up example-full.yaml -y`
`ray exec example-full.yaml "watch ray status"`

example-full.yaml used: https://gist.github.com/xwjiang2010/772b799a70b0902ce78acce30574f40a

Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [X] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by wuisawesome** [comment]
Yeah, this is the intended behavior, since setting `--num-cpus` would leave the autoscaler in a dangerously inconsistent state. We should add a good error message that says to set the `CPU` field in `node_config: {resources: {CPU: ...}}` instead.

**Comment by xwjiang2010** [comment]
Got it! Thanks for the info.

**Comment by wuisawesome** [comment]
I'm going to reopen this, because if you were confused by this, then we should fix the message since others would be confused too

**Comment by xwjiang2010** [comment]
Sounds good. Just curious, why would it "leave autoscaler in a dangerously inconsistent state"?

**Comment by wuisawesome** [comment]
Bumping to P2, but we should consider whether we even want the resource overrides feature in its current state in ray 2.0

**Comment by cszhu** [comment]
Hello! This P2 issue has seen no activity in at least 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
